### PR TITLE
Disable eth sign

### LINKED ISSUE #5676: Disable eth_sign by default
**Background**

From Harry:
After looking at various phishing kits ([and de-obfuscating](https://twitter.com/sniko_/status/1587882460976713728)), there are enormous trends between different kits that they all abuse `eth_sign` to ask users to inadvertently sign a transaction (as MetaMask UI is opaque to an on-chain transaction being signed). You can read more about the attack [here](https://officercia.medium.com/how-hackers-may-steal-your-ethers-and-why-does-eth-sign-function-matter-69497ee11593).

@409H'd estimate that the abuse of `eth_sign` is near equal to the abuse of Wyvern and Seaport contracts in a manner to steal NFTs through creating orders for 0 eth to a specific beneficiary.

In short: `eth_sign` has been consistently _abused_ by phishers/scammers to steal funds and the case may be that only power-users who use very legacy dapps may need to still use eth_sign.

- Disable `eth_sign` by default - throw an ethErrors.rpc.methodNotFound exception
- Allow the user to enable `eth_sign` through "Advanced Settings"

RPC Method preferences are stored within `rpcMethodPreferences` in the store state, allow for us to develop this further if someone wants to lock down rpc calls into MetaMask.

### PR DESCRIPTION
**Pull request by shanejonas**
**Description**
Disables eth_sign by default. Users must enable `eth_sign` through "Advanced Settings"

See #5676 for more information.

**Screenshots/Recordings**

https://user-images.githubusercontent.com/364566/220941355-6f2b8317-2b2f-4e0a-900e-46afb02fbdb0.mov

Closes #5676

### CODE REVIEW

**Comment by shanejonas** [comment]
Couldn't find any existing tests for features of the `AdvancedSettings` component to add to

**Comment by BelfordZ** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ios/Podfile.lock [hunk 1]
```diff
@@ -777,7 +777,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   boost: a7c83b31436843459a1961bfd74b96033dc77234
-  Branch: 74cc856025984f691833c8fa332834ac38a0cf4e
+  Branch: 4ac024cb3c29b0ef628048694db3c4cfa679beb0
```
Comment (blackdevelopa) [inline ios/Podfile.lock:780]: nitpicking: Should we include this to the PR?

**Review by blackdevelopa — APPROVED** [review]
This looks good to me. Just some small changes: one nitpick, cleaner lint (`yarn lint:fix`) and snapshot test (`yarn run test:unit:components-views -u`)

**Comment by Cal-L** [comment]
Let's also include translations either in this PR or follow up with a translations PR. Shoot me a message if you need help - I can walk you through it

**Comment by DigitalsupportTeam** [comment]
We'd be happy to look into this for you
kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

**Pull request closed without merging**